### ISSUE TITLE
Upgrade from 12 to 13

### ISSUE THREAD

**Issue by Eralmidia** [IgniteUI/igniteui-angular#11528]
## Question  
I'm trying to upgrade from igx 12.2 to 13.1, and like with our upgrade from 11 til 12, I'm having issues with the migrations. I'm getting the following error:
![image](https://user-images.githubusercontent.com/12936286/167841794-14b9260a-c42b-47ec-8c8f-67cc50238352.png)

Any idea what could be causing this? Also, this error seems to be for 13 til 13.1. Does that mean that the 12.2 to 13 was upgraded successfully, or does the upgrade roll back everything if it fails?

**Comment by Eralmidia** [comment]
Never mind that last question. I see that there are changes to dozens of components, so the 12.2 to 13 seems to be fine. Is it viable to just check https://www.infragistics.com/products/ignite-ui-angular/angular/components/general/update-guide#from-122x-to-130x manually?

**Comment by Lipata** [comment]
@Eralmidia we have a PR for that #11512, which aligns our migrations with latest changes from typescript.

**Comment by Eralmidia** [comment]
Dev dependancies you mean?

"@angular-devkit/schematics-cli": "0.1000.6",
    "@angular/animations": "12.2.13",
    "@angular/cdk": "12.2.12",
    "@angular/common": "12.2.13",
    "@angular/compiler": "12.2.13",
    "@angular/core": "12.2.13",
    "@angular/forms": "12.2.13",
    "@angular/material": "12.2.12",
    "@angular/platform-browser": "12.2.13",
    "@angular/platform-browser-dynamic": "12.2.13",
    "@angular/router": "12.2.13",
    "@infragistics/igniteui-angular": "12.3.2",
    "igniteui-angular-charts": "12.1.1",
    "igniteui-angular-core": "12.1.1",
    
    I ran the angular upgrade guide (https://update.angular.io/?v=12.0-13.0) first yes, so Angular and angular-cli was upgrade before upgrading igx.
    
    Type script was at 4.3, but the angular upgrade seems to have brought this up to 4.6.

**Comment by Lipata** [comment]
@Eralmidia what is the output, when you execute "ng --version"?

**Comment by Lipata** [comment]
@Eralmidia can you try with the latest [13.1.15](https://github.com/IgniteUI/igniteui-angular/releases/tag/13.1.15) patch and see if you still have that error?

**Comment by Eralmidia** [comment]
@Lipata Thanks, the migrations seems to be working now. Have to clean up a lot of these manually before actually testing tho
![image](https://user-images.githubusercontent.com/12936286/169458976-9a36c746-0118-47a0-b2e3-f99c04ff70f5.png)

**Comment by jackofdiamond5** [comment]
Hello, @Eralmidia I've logged an issue for the problem that you're describing (https://github.com/IgniteUI/igniteui-angular/issues/11593). Has everything else went smoothly while migrating or have you experienced any other issues?

**Comment by Eralmidia** [comment]
@jackofdiamond5 Yes, it seems to have gone fine, but I've been fixing misc issues with the build (both igx and other frameworks). The last issue I'm left with is the following:
![image](https://user-images.githubusercontent.com/12936286/169524428-987918c8-b6af-41c8-a06a-59b05154f27f.png)
This method seems to no longer be available.

**Comment by Eralmidia** [comment]
@jackofdiamond5 In addition I have found one smaller issue. Guess this might be one for @simeonoff? After the upgrade, it seems like the $item-hover-background theme variable for the Navdrawer is no longer respected. I have set up my theme like this:

![image](https://user-images.githubusercontent.com/12936286/169751765-8142b0f8-6f60-40c9-8bdd-2d8001988ea8.png)

The dark pallette 400 refers to this:
![image](https://user-images.githubusercontent.com/12936286/169751832-6234661d-a2f2-4e3d-bcc1-f3fd051c329f.png)

Still when I hover a nawdrawer item it now applies a much lighter color:
![image](https://user-images.githubusercontent.com/12936286/169751960-f28e84e2-74ab-403b-90ad-24d9c19ef9a6.png)
![image](https://user-images.githubusercontent.com/12936286/169752078-cee6c95d-e495-4c86-ae48-8219545281c7.png)
